src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/Reader/NativeWrapper.cs

```diff
@@ -1161,13 +1181,18 @@ private static object ConvertToObject(UnsafeNativeMethods.EvtVariant val)
                 case (int)UnsafeNativeMethods.EvtVariantType.EvtVarTypeSid:
                     return (val.SidVal == IntPtr.Zero) ? null : new SecurityIdentifier(val.SidVal);
                 case (int)UnsafeNativeMethods.EvtVariantType.EvtVarTypeGuid:
-                    return (val.GuidReference == IntPtr.Zero) ? Guid.Empty : Marshal.PtrToStructure<Guid>(val.GuidReference);
+#if NET
```

Comment:
This is unnecessary and actually wrong.
- This is Windows specific code. Windows is always going to be little endian system.
- In the hypothetical case Windows run on a big endian system, System.Guid would still have to match the GUID returned by the Windows API and no endianness adjustment would be required.

Reply:
addressed d4221e6fc8f4226f3ddb91f7aad9bc38f154d1d1